File: RemoteBees-iOS/RemoteBeesService/API/BeehiveService.swift
/***************************************************************************
 * This source file is part of the RemoteBees open source project.         *
 *                                                                         *
 * Licensed under the MIT License. See LICENSE for license information     *
 ***************************************************************************/

import UIKit
import PromiseKit

public enum JobSearchType {
    case category(String)
    case companyName(String)
    case listing(String)
}

public protocol BeehiveService {
    
    func getAllJobs() -> Promise<[Job]>

    func getJob(_ id: Int) -> Promise<Job?>

    func searchJobs(by type: JobSearchType) -> Promise<[Job]>

    func getCompanyLogo(_ id: Int) -> Promise<UIImage?>

}

File: RemoteBees-Android/RemoteBeesService/src/main/java/dev/beehive/remotebees/service/api/BeehiveService.kt
package dev.beehive.remotebees.service.api

import android.graphics.Bitmap
import com.inmotionsoftware.promisekt.Promise
import dev.beehive.remotebees.service.domain.Job

sealed class JobSearchType {
    data class Category(val term: String): JobSearchType()
    data class CompanyName(val term: String): JobSearchType()
    data class Listing(val term: String): JobSearchType()
}

interface BeehiveService {

    fun getAllJobs(): Promise<List<Job>>

    fun getJob(id: Int): Promise<Job?>

    fun searchJobs(type: JobSearchType): Promise<List<Job>>

    fun getCompanyLogo(id: Int): Promise<Bitmap?>

}
